File: rust/src/bito.rs
use std::time::Duration;
use tokio::time::sleep;

pub struct BitoAgent {
    pub model_profile: String,
    pub max_context: usize,
}

impl BitoAgent {
    pub fn new() -> Self {
        Self {
            model_profile: "ADVANCED".to_string(),
            max_context: 240_000,
        }
    }

    pub fn set_model_profile(&mut self, profile: &str) {
        if profile == "BASIC" {
            self.max_context = 40_000;
        } else {
            self.max_context = 240_000;
        }
        self.model_profile = profile.to_string();
        println!("Bito Model Profile set to: {} (Limit: {})", self.model_profile, self.max_context);
    }

    pub async fn inject_prompt_macro(&self, template: &str, file_content: &str) -> Result<String, String> {
        sleep(Duration::from_millis(50)).await;

        let result = template.replace("{{%input%}}", file_content);

        if result.len() > self.max_context {
            return Err(format!("Injected prompt exceeds maximum context length of {}", self.max_context));
        }

        Ok(result)
    }
}

File: go/internal/agent/bito.go
package agent

import (
	"context"
	"fmt"
	"strings"
	"time"
)

// BitoAgent represents the Go port of the Bito CLI functionality
type BitoAgent struct {
	BaseAgent
	ModelProfile string
	MaxContext   int
}

func NewBitoAgent() *BitoAgent {
	return &BitoAgent{
		ModelProfile: "ADVANCED",
		MaxContext:   240000,
	}
}

// SetModelProfile changes the model restriction tier (BASIC vs ADVANCED)
func (a *BitoAgent) SetModelProfile(profile string) {
	if profile == "BASIC" {
		a.MaxContext = 40000
	} else {
		a.MaxContext = 240000
	}
	a.ModelProfile = profile
	fmt.Printf("Bito Model Profile set to: %s (Limit: %d)\n", profile, a.MaxContext)
}

// InjectPromptMacro injects the provided file content into the prompt template
func (a *BitoAgent) InjectPromptMacro(ctx context.Context, template string, fileContent string) (string, error) {
	time.Sleep(50 * time.Millisecond)

	result := strings.ReplaceAll(template, "{{%input%}}", fileContent)

	if len(result) > a.MaxContext {
		return "", fmt.Errorf("injected prompt exceeds maximum context length of %d", a.MaxContext)
	}

	return result, nil
}
